components/match2/wikis/valorant/match_group_input_custom.lua

```diff
@@ -484,166 +183,68 @@ end
 -- DUMMY_MAP_NAME needs the match the default value in Template:Map
 ---@param map table
 ---@return boolean
-function mapFunctions.discardMap(map)
-	if map.map == DUMMY_MAP_NAME then
-		return true
-	else
-		return false
-	end
+function MapFunctions.keepMap(map)
+	return map.map ~= DUMMY_MAP
 end
 
 ---@param map table
 ---@return table
-function mapFunctions.getExtraData(map)
-	map.extradata = {
+function MapFunctions.getExtraData(map, participants)
+	local extraData = {
 		comment = map.comment,
 		t1firstside = map.t1firstside,
 		t1halfs = {atk = map.t1atk, def = map.t1def, otatk = map.t1otatk, otdef = map.t1otdef},
 		t2halfs = {atk = map.t2atk, def = map.t2def, otatk = map.t2otatk, otdef = map.t2otdef},
 	}
-	return map
-end
 
----@param map table
----@return table
-function mapFunctions.getScoresAndWinner(map)
-	map.scores = {}
-	local indexedScores = {}
-	for scoreIndex = 1, MAX_NUM_OPPONENTS do
-		-- read scores
-		local score = map['score' .. scoreIndex]
-		if map['t'.. scoreIndex ..'atk'] or map['t'.. scoreIndex ..'def'] then
-			score = (tonumber(map['t'.. scoreIndex ..'atk']) or 0)
-					+ (tonumber(map['t'.. scoreIndex ..'def']) or 0)
-					+ (tonumber(map['t'.. scoreIndex ..'otatk']) or 0)
-					+ (tonumber(map['t'.. scoreIndex ..'otdef']) or 0)
-		end
-		local obj = {}
-		if not Logic.isEmpty(score) then
-			if TypeUtil.isNumeric(score) then
-				obj.status = 'S'
-				obj.score = score
-			elseif Table.includes(ALLOWED_STATUSES, score) then
-				obj.status = score
-				obj.score = -1
-			end
-			table.insert(map.scores, score)
-			indexedScores[scoreIndex] = obj
-		else
-			break
-		end
+	for key, participant in pairs(participants) do
+		local opponentIdx, playerIdx = unpack(mw.text.split(key, '_', true))
+		extraData['t' .. opponentIdx .. 'p' .. playerIdx] = participant.player
+		extraData['t' .. opponentIdx .. 'p' .. playerIdx .. 'agent'] = participant.agent
 	end
 
-	map = CustomMatchGroupInput.getResultTypeAndWinner(map, indexedScores)
-
-	return map
+	return extraData
 end
 
 ---@param map table
+---@param opponentCount integer
 ---@return table
-function mapFunctions.getParticipantsData(map)
-	local participants = map.participants or {}
+function MapFunctions.getParticipants(map, opponentCount)
```

Comment:
same question as on dota2

Reply:
The ID between opponent layer and game layer may not be the same on valorant, ie match.opponent[1].players[4] and match.games[1].participant['1_4'] may not be the same 

Not sure if that's a problem

Reply:
Edit: never, specs says they should be linked

Reply:
However they are not done so on live, I think we can do an additional pass afterwards to have some standardized functions that does this

Reply:
it is against what we describe in the docs
also it will also cause issues with e.g. the games table luckelucky worked on

https://liquipedia.net/commons/Help:LiquipediaDB/Match
![image](https://github.com/user-attachments/assets/5aeab088-4106-4dd3-8542-e1570198d9e2)

i mean we can leave it oos for now (since the old implementation doies the same fault) but eventually we should fix it

Reply:
Yeah that's what my "nevermind specs say they should be linked" was about